tests/unit_tests/udf_communication/peer_communication/test_payload_handler.py

```diff
@@ -0,0 +1,100 @@
+import dataclasses
+from typing import Union
+from unittest.mock import create_autospec, MagicMock, call
+
+import pytest
+from polyfactory.factories.pydantic_factory import ModelFactory
+
+from exasol_advanced_analytics_framework.udf_communication import messages
+from exasol_advanced_analytics_framework.udf_communication.peer_communicator.payload_handler import PayloadHandler
+from exasol_advanced_analytics_framework.udf_communication.peer_communicator.payload_receiver import PayloadReceiver
+from exasol_advanced_analytics_framework.udf_communication.peer_communicator.payload_sender import PayloadSender
+from exasol_advanced_analytics_framework.udf_communication.socket_factory.abstract import Frame
+from tests.mock_cast import mock_cast
+
+
+@dataclasses.dataclass
+class TestSetup:
+    payload_sender_mock: Union[MagicMock, PayloadSender]
+    payload_receiver_mock: Union[MagicMock, PayloadReceiver]
+    payload_handler: PayloadHandler
+
+    def reset_mock(self):
+        self.payload_sender_mock.reset_mock()
+        self.payload_receiver_mock.reset_mock()
+
+
+def create_test_setup() -> TestSetup:
+    payload_sender_mock = create_autospec(PayloadSender)
+    payload_receiver_mock = create_autospec(PayloadReceiver)
+    payload_handler = PayloadHandler(
+        payload_sender=payload_sender_mock,
+        payload_receiver=payload_receiver_mock
+    )
+    return TestSetup(
+        payload_handler=payload_handler,
+        payload_sender_mock=payload_sender_mock,
+        payload_receiver_mock=payload_receiver_mock
+    )
+
+
+def test_init():
+    test_setup = create_test_setup()
+    assert test_setup.payload_receiver_mock.mock_calls == [] and \
+           test_setup.payload_sender_mock.mock_calls == []
+
+
+def test_try_send():
+    test_setup = create_test_setup()
```

Comment:
does it make sense to create a fixture for that like:
```
@pytest.fixture
def test_setup():
    test_setup = create_test_setup()
    test_setup.reset_mock()
    return test_setup
```

Reply:
puh, good question. we need the reset_mock in every test except `test_init`, but on the other side it is only function call per test. Maybe more like extract a function for it.